Drivers/Src/BatteryBalancing.c

```diff
@@ -0,0 +1,78 @@
+
+#include "LTC6811.h"
+#include "Voltage.h"
+#include "BatteryBalancing.h"
+#include "config.h"
+#include "Current.h"
+
+#define CHARGING_TOLERANCE 0
+
+static uint16_t Balancing_Init(void) {
+	uint16_t minVoltage = 0;
+	if (Current_CheckStatus(0) == 1){
+		return;//If charging, return
+	}
+	Voltage_UpdateMeasurements();
+	minVoltage = Voltage_GetModuleMillivoltage(0);
+	for (uint8_t i = 1; i < NUM_BATTERY_MODULES; i++){
+		if(Voltage_GetModuleMillivoltage(i) < minVoltage) {
+			minVoltage = Voltage_GetModuleMillivoltage(i);//Find minimum voltage
+		}
+	}
+	return minVoltage;
+}
+	
+void Balancing_Balance(cell_asic Minions[]){ 
+	uint16_t lowest = Balancing_Init(); //get lowest voltage 
+	Voltage_UpdateMeasurements();
```

Comment:
We should not be updating Voltage measurements in the battery balancing functions